### PR TITLE
staging monitor extension

### PR DESCRIPTION
**Pull request by mgcam**

### CODE REVIEW

File: bin/staging_area_monitor [hunk 1]
```diff
@@ -32,32 +39,49 @@ my @area_list = $monitor->validate_areas(@ARGV);
 
 croak 'No valid arguments' if scalar @area_list == 0;
 
-my $iteration = 0;
+# In production dev variable is not set
+my $dry_run = $ENV{'dev'};
```
Comment (dkj) [inline bin/staging_area_monitor]: Intentional loss of
 ```$ENV{dev} and ( $ENV{dev} ne 'live' )```
logic?

File: lib/Monitor/RunFolder/Staging.pm [hunk 2]
```diff
@@ -275,8 +331,20 @@ sub _change_group {
 
     my $gid = getgrnam($group);
     chown -1, $gid, $directory;
+    # If needed, set group sticky bit
```
Comment (dkj) [inline lib/Monitor/RunFolder/Staging.pm]: Terminology is non-obvious and I've got this wrong in the past. We're not using the "sticky" bit:
S_ISGID: take a new file's group from parent directory,
S_ISVTX: sticky bit (restricted deletion flag)

File: bin/staging_area_monitor (hunk 1 above)
Comment (mgcam) [inline bin/staging_area_monitor]: > Intentional loss of
> |$ENV{dev} and ( $ENV{dev} ne 'live' )|
> logic?
>
Yes. This dry_run facility is, unfortunately, not clean, ie some tags 
are still set for the run. Therefore, if one wants to test the whole 
script, either more work has to be done or dev=dev still should be used. 
I was using it with dev=live and an early return so most of the code is 
not executed, but I still get true representation of the status of the 
file system in conjunction with the run status. Can put some extra time 
into setting tags conditionally, then we'll have a true dry_run mode




-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.

File: lib/Monitor/RunFolder/Staging.pm (hunk 2 above)
Comment (mgcam) [inline lib/Monitor/RunFolder/Staging.pm]: Added a better comment
> > +    # If needed, set group sticky bit



-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.

**Pull request merged**